MSBuildTasks/FilterTestingConfigurations.cs

```diff
@@ -0,0 +1,108 @@
+{
+  public class FilterTestingConfigurations : Task
+  {
+    private readonly ITaskLogger _logger;
+    [Required] public ITaskItem[] Input { get; set; }
+    [Required] public ITaskItem[] ValidPlatforms { get; set; }
+    [Required] public ITaskItem[] ValidDatabaseSystems { get; set; }
+    [Required] public ITaskItem[] ValidBrowsers { get; set; }
+
+    [Output] public ITaskItem[] Output { get; set; }
+
+    public FilterTestingConfigurations ()
+    {
+      _logger = new TaskLogger (Log);
+    }
+
+    public FilterTestingConfigurations (ITaskLogger logger)
+    {
+      _logger = logger;
+    }
+
+    public override bool Execute ()
+    {
+      var validInputs = Input
+          .Where (HasValidPlatform)
+          .Where (HasValidDatabaseSystem)
+          .Where (HasValidBrowser);
+      Output = validInputs.ToArray();
+
+      _logger.LogMessage (
+          @"The following test configurations were ignored:
+{0}",
+          string.Join (Environment.NewLine, Input.Except (Output).Select (GetConfigurationString)));
+
+      var uniqueInputAssemblyNames = GetUniqueAssemblyNames (Input);
+      var uniqueOutputAssemblyNames = GetUniqueAssemblyNames (Output);
+      var untestedAssemblyNames = uniqueInputAssemblyNames.Where (assemblyName => !uniqueOutputAssemblyNames.Contains (assemblyName));
+      foreach (var assemblyName in untestedAssemblyNames)
+      {
+        _logger.LogWarning ($"All testing configurations in {assemblyName} were ignored.");
+      }
+
+      return true;
+    }
+
+    private string GetConfigurationString (ITaskItem taskItem)
+    {
+      var browser = taskItem.GetMetadata (TestingConfigurationMetadata.Browser);
+      var platform = taskItem.GetMetadata (TestingConfigurationMetadata.Platform);
+      var databaseSystem = taskItem.GetMetadata (TestingConfigurationMetadata.DatabaseSystem);
+      var executionRuntime = taskItem.GetMetadata (TestingConfigurationMetadata.ExecutionRuntime);
+      var configurationID = taskItem.GetMetadata (TestingConfigurationMetadata.ConfigurationID);
+      return $"{taskItem.ItemSpec}: {browser}, {databaseSystem}, {platform}, {executionRuntime}, {configurationID}";
+    }
+
+    private bool HasValidPlatform (ITaskItem item)
+    {
+      return ValidPlatforms.Select (i => i.ItemSpec).Contains (item.GetMetadata (TestingConfigurationMetadata.Platform));
+    }
+
+    private bool HasValidDatabaseSystem (ITaskItem item)
+    {
+      var database = item.GetMetadata (TestingConfigurationMetadata.DatabaseSystem);
+      if (string.Equals (database, "NoDb", StringComparison.OrdinalIgnoreCase) && !ValidDatabaseSystems.Any())
+        return true;
+
+      return ValidDatabaseSystems.Select (i => i.ItemSpec).Contains (item.GetMetadata (TestingConfigurationMetadata.DatabaseSystem));
+    }
+
+    private bool HasValidBrowser (ITaskItem item)
+    {
+      var browser = item.GetMetadata (TestingConfigurationMetadata.Browser);
+      if (browser == "NoBrowser" && !ValidBrowsers.Any())
```

Comment:
We debated this and only added the case insensitivity because we predicted that "No**DB**" vs "No**Db**" would be a problem in the future. We could make all of it case insensitive, though.

Reply:
Mhmm, valid problem, still seems random. We either make everything case insensitive or none for predictable behaviour. Maybe check what the default msbuild behaviour is with case sensitivity and use that. If MsBuild is case insensitive, we should probably also use it for the Check if it is a valid browser and the other stuff?

Reply:
Yes, this is an interesting issue. MSbuild seemingly treats everything as case-insensitive, but since we can define the wanted behavior here, we're not quite sure what to do and will check with MK.

Reply:
Spoke to MK, since MSBuild is per default case-insensitive, we should in fact try to mirror that behavior in our C# tasks.

Reply:
Well at least one out of three was correct :D